lib/tasks/data_migration.rake

```diff
@@ -79,4 +79,62 @@ namespace :data_migration do
     puts "Associating protocol for #{organization.facility_groups.count} facility groups"
     organization.facility_groups.update(protocol: protocol)
   end
+
+  desc "Create automatic appointment for defaulters so that they show up in the Overdue List"
+  task create_automatic_appointment_for_defaulters: :environment do
+    last_bp_older_than_one_month = ->(p) { p.latest_blood_pressure.blank? || p.latest_blood_pressure.device_created_at < 1.month.ago }
+
+    patient_is_hypertensive = ->(p) { p.latest_blood_pressure.present? && (p.latest_blood_pressure.systolic > 140 || p.latest_blood_pressure.diastolic < 90) }
+
+    patient_has_medical_history = ->(p) { (p.medical_history.present? && (p.medical_history.prior_stroke == "yes" || p.medical_history.prior_heart_attack == "yes" || p.medical_history.chronic_kidney_disease == "yes" ||
+      p.medical_history.diabetes == "yes")) }
+
+    defaulters = Patient.select do |p|
+      p.appointments.count == 0 &&
+        last_bp_older_than_one_month.call(p) &&
+        (patient_is_hypertensive.call(p) || patient_has_medical_history.call(p) || p.prescription_drugs.present?)
+    end
+
+    puts "Found #{defaulters.count} defaulters"
+
+    if defaulters.blank? || defaulters.empty?
+      puts "No defaulters found. Aborting task."
+      return
+    end
+
+    puts "Processing #{defaulters.count} defaulters..."
+
+    processed_defaulters_count = 0
+    unprocessed_or_errored_defaulters_count = 0
+
+    defaulters.each do |defaulter|
+      if defaulter.latest_blood_pressure.blank?
+        puts "Failed to find latest Blood Pressure reading for defaulter #{defaulter.id}. Skipping patient..."
+        unprocessed_or_errored_defaulters_count += 1
+        next
+      end
+
+      latest_bp_facility_id = defaulter.latest_blood_pressure.facility_id
+      app_creation_time = Time.now
+      app_scheduled_date = 32.days.ago # to trigger overdue-ness
```

Comment:
Maybe just do this `40.days.ago` or `patient.latest_blood_pressure.device_created_at + 1.month`?

Reply:
Let's check when to create the appointment.

Reply:
Changed it to use `40 days ago`.